### PR TITLE
Implement `tag_data` for `ak.read_parquet` on the whole data set

### PR DESCRIPTION
**Pull request by jabraham17**
Implements `tag_data` for `ak.read_parquet` for the whole dataset, which was missing before

I also added a whole bunch of tests for `tag_data`, since there were none before

### CODE REVIEW

File: src/ParquetMsg.chpl [hunk 1]
```diff
@@ -1030,19 +1030,21 @@ module ParquetMsg {
       pqErr.parquetError(getL(), getR(), getM());
     }
 
-    pqLogger.debug(getM(),getR(),getL(), "will try tagging data");
-
-    // If tagging is turned on, tag the data
-    if tagData {
-      throw new NotImplementedError("Reading all columns while tagging " +
```
Comment (benharsh) [inline src/ParquetMsg.chpl:1037]: Should we recreate this error in the non-optimizable case?

Comment (jabraham17) [reply src/ParquetMsg.chpl:1037]: No need. In the non-optimized case we fall back to an implementation that tags the data

**Review by benharsh — APPROVED** [review]
